integration/service/create_test.go

```diff
@@ -440,7 +462,7 @@ func TestCreateServiceSysctls(t *testing.T) {
 	}
 }
 
-func serviceRunningTasksCount(client client.ServiceAPIClient, serviceID string, instances uint64) func(log poll.LogT) poll.Result {
+func serviceRunningTasksCount(client client.ServiceAPIClient, serviceID string, instances, maxReplicas uint64) func(log poll.LogT) poll.Result {
```

Comment:
Do we need to change this?
I'm thinking we could do two samples in the test, 1 to make sure that the expected number (max replicas) is reached, and another to make sure that we never get passed max replicas.

Granted checking to make sure that we never go passed the max is racey by definition (e.g. a bad implementation could just have taken longer than expected to place the task)... but I'm not sure how else we'd do that.

Reply:
@cpuguy83 there is quite many test cases on swarmkit side: https://github.com/docker/swarmkit/blob/master/manager/scheduler/scheduler_test.go#L3266-L3517 is there really point to duplicate them here? IMO, swarmkit is responsible of this logic and on my side we are interested that API works as expected.

Reply:
If we don't want to test that, then can't we only pass MaxReplicas in as `instances` and forgo adding the new argument?

Reply:
@cpuguy83 changed like that